### PR TITLE
Fix `sts_client_profile` errors.

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/23672|

## Description

Review and fix subscriber-type related classes in the AWS module.

### CODE REVIEW

**Review by GGP1 — CHANGES_REQUESTED** [review]
Changes LGTM, but there's an integration test case related to SQS queues that is [failing](https://github.com/wazuh/wazuh/actions/runs/9472259170/job/26099370778?pr=24050). It passed in master before, so we should fix it.

```console
assert log_monitor.callback_result is not None, ERROR_MESSAGE['failed_sqs_message_retrieval']
E       AssertionError: The AWS module did not retrieve the expected message from the SQS Queue
E       assert None is not None
E        +  where None = <wazuh_testing.tools.monitors.file_monitor.FileMonitor object at 0x7f3764c1b190>.callback_result
```

**Review by GGP1 — CHANGES_REQUESTED** [review]
There are some conflicts that should be resolved before merging, please make sure that the checks run after these changes.

**Comment by RamosFe** [comment]
## Update

After fixing the merge conflicts and running again the [IT](https://github.com/wazuh/wazuh/actions/runs/9875706682/job/27273076544?pr=24050), no errors related to the changes made in the Issue were found.

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: wodles/aws/services/aws_service.py [hunk 1]
```diff
@@ -55,10 +55,10 @@ class AWSService(wazuh_integration.WazuhAWSDatabase):
         The desired duration of the session that is going to be assumed.
     """
 
-    def __init__(self, reparse: bool, profile: str, iam_role_arn: str,
-                 service_name: str, only_logs_after: str, account_alias: str, region: str,
-                 db_table_name: str = DEFAULT_TABLENAME, discard_field: str = None,
-                 discard_regex: str = None, sts_endpoint: str = None, service_endpoint: str = None,
+    def __init__(self, reparse: bool, profile: str, iam_role_arn: str,service_name: str, only_logs_after: str,
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply wodles/aws/services/aws_service.py]: Fixed in commit a7d2afc59558d1e6ff7e2111ebebb0bf3ca0bceb.

**Review by fdalmaup — APPROVED** [review]
LGTM

**Comment by RamosFe** [comment]
## Update

Due to conflict errors when rebasing the branch in an attempt to `amend` the commits and sign them in the process, I finally squashed the commits and signed them. I will wait for the AWS Integration test results before sending it to review.

**Pull request merged**